json-definitions/v3/tech-record/get/hgv/complete/index.json

```diff
@@ -1074,6 +1084,33 @@
           ]
         }
       },
+      {
+        "if": {
+          "properties": {
+            "techRecord_adrDetails_permittedDangerousGoods": {
+              "type": "array",
+              "contains": {
+                "type": "string",
+                "not": {
+                  "enum": [
+                    "Explosives (type 3)"
+                  ]
+                }
+              }
+            }
+          },
+          "required": [
+            "techRecord_adrDetails_permittedDangerousGoods"
+          ]
+        },
+        "then": {
+          "not": {
+            "required": [
+              "techRecord_adrDetails_bodyDeclaration_type"
+            ]
+          }
+        }
+      },
```

Comment:
Looking at this, it makes sense to me but when looking at the TS types it doesn't seem to include this validation. At what point will this if check be applied when using the type? I'm also wondering if this should be, moved into one of the services; seems more like business logic

Reply:
Types don't get validation, that's not what they're for

This check will be applied when the validation schema is ran, the types definitions also create JSON schemas which we use. We have lots of validation rules using these. I would not recommend adding these into the endpoints as why is this validation different to the rest? Also, it would have to be on multiple endpoints, which is why we have it distributed here.